contracts/utils/TimeLock.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { ITimeLock } from '../interfaces/ITimeLock.sol';
+
+/**
+ * @title TimeLock
+ * @author Kiryl Yermakou
+ * @dev A contract that enables function execution after a certain time has passed.
+ * Implements the ITimeLock interface.
+ */
+contract TimeLock is ITimeLock {
+    bytes32 internal constant PREFIX_TIME_LOCK = keccak256('time-lock');
+
+    uint256 internal immutable MINIMUM_TIME_LOCK_DELAY;
```

Comment:
Use camel case for naming. I think the convention is that only constants (not immutables) should use capital snake case.